### PR TITLE
Tweaks to add remote selenium driver

### PR DESCRIPTION
**Pull request by rapatel0**
The changes enable usage of `driver.remote()` in selenium for remote execution.

### CODE REVIEW

File: browserpilot/agents/gpt_selenium_agent.py [hunk 1]
```diff
@@ -116,7 +119,6 @@ def __init__(
 
         """Set up the driver."""
         _chrome_options = webdriver.ChromeOptions()
-        _chrome_options.add_argument(f"user-data-dir={user_data_dir}")
```
Comment (handrew) [inline browserpilot/agents/gpt_selenium_agent.py:119]: Why take this out? If it's local you might still want to use the `user_data_dir`. How is that handled remotely? 

Maybe the right way to go is to check if the `remote_url` is existent, and if not, then we add the user data dir?

File: browserpilot/agents/gpt_selenium_agent.py [hunk 2]
```diff
@@ -89,7 +91,7 @@ def __init__(
             or instruction_output_file.endswith(".json")
         ), "Instruction output file must be a YAML or JSON file or None."
         assert (
-            chromedriver_path is not None
+            (chromedriver_path is not None) or (remote_url is not None)
```
Comment (handrew) [inline browserpilot/agents/gpt_selenium_agent.py:94]: Probably have to check if it's one or the other, right?

File: browserpilot/agents/gpt_selenium_agent.py (hunk 1 above)
Comment (handrew) [reply browserpilot/agents/gpt_selenium_agent.py:119]: Oh, no, I see it later in the diff.

**Comment by handrew** [comment]
I think it looks mostly good to me, except the one nit on the assertion. 

I'll have to test it myself some time—I didn't know you could do this with Selenium!—but otherwise thank you for the contribution and I'll try to merge it quickly.

**Comment by rapatel0** [comment]
Hold up. Keep this in pull. I see a few issues that I'd like to merge as well.

**Comment by rapatel0** [comment]
If you want to test

```
docker run -d -p 4444:4444 -p 7900:7900 -e SE_NODE_MAX_SESSIONS=8 --shm-size="4g" --name="sel" selenium/standalone-chrome
```

**Comment by rapatel0** [comment]
All good now. Tested with my workflow. 

If you guys want to test:
```
docker run -d -p 4444:4444 -p 7900:7900 -e SE_NODE_MAX_SESSIONS=8 --shm-size="4g" --name="SEL" selenium/standalone-chrome
```

```d
ef run_agent(instructions, proxy, config_arguements):

    agent = GPTSeleniumAgent(instructions, 
                             remote_url=http://localhost:4444,
                             chrome_options=chrome_options, 
                             user_data_dir=config_arguements.user_data_dir,
                             headless=True)
    agent.run()

**Comment by rapatel0** [comment]
>I'll have to test it myself some time—I didn't know you could do this with Selenium!—but otherwise thank you for the contribution and I'll try to merge it quickly.

Yeah maintaining selenium, chrome, and chromedriver is a pain. TThis is easier to just fireoff and run from a remote location.

**Comment by handrew** [comment]
Taking a look now.

**Comment by handrew** [comment]
> All good now. Tested with my workflow.
> 
> If you guys want to test
> 
> **Selenium Remote Driver**
> 
> ```
> docker run -d -p 4444:4444 -p 7900:7900 -e SE_NODE_MAX_SESSIONS=8 --shm-size="4g" --name="SEL" selenium/standalone-chrome
> ```
> 
> **Calling the agent**
> 
> ```python
> def run_agent(instructions, config_arguements):
> 
>     agent = GPTSeleniumAgent(instructions, 
>                              remote_url=http://localhost:4444,
>                              user_data_dir=config_arguements.user_data_dir,
>                              headless=True)
>     agent.run()
> ```

Hey @rapatel0 I tried this and got the following error: `selenium.common.exceptions.SessionNotCreatedException: Message: Could not start a new session. Error while creating session with the driver service. Stopping driver service: Could not start a new session. Response code 500. Message: unknown error: cannot create default profile directory`.

Once I can replicate it working then I'll merge and update the package.

**Comment by handrew** [comment]
Thanks @rapatel0, just updated the package. 0.2.51. Appreciate the update, this was cool to try out.

**Pull request merged**